index.js

```diff
@@ -359,6 +360,56 @@ module.exports = class Autobase extends ReadyResource {
     if (this.localWriter && this._ackInterval) this._startAckTimer()
   }
 
+  async _onreindexing (record) {
+    const { key, length } = messages.Checkout.decode({ buffer: record, start: 0, end: record.byteLength })
+    const encryptionKey = this._viewStore.getBlockKey(this._viewStore.getSystemCore().name)
+    const core = this.store.get({ key, encryptionKey, isBlockKey: true }).batch({ checkout: length, session: false })
+
+    const base = this
+    const system = new SystemView(core, length)
+    await system.ready()
+
+    const indexerCores = []
+    for (const { key } of system.indexers) {
+      const core = this.store.get({ key, compat: false, valueEncoding: messages.OplogMessage, encryptionKey: this.encryptionKey })
+      indexerCores.push(core)
+    }
+
+    await system.close()
+
+    for (const core of indexerCores) tail(core).catch(safetyCatch)
+
+    async function onsyskey (key) {
+      for (const core of indexerCores) await core.close()
+      if (key === null || !base.reindexing) return
+      base.initialFastForward(key, DEFAULT_FF_TIMEOUT)
+    }
+
+    async function tail (core) {
+      await core.ready()
+
+      while (base.reindexing) {
+        const seq = core.length - 1
+        const blk = seq >= 0 ? await core.get(seq) : null
+        if (blk && blk.version === 1) {
```

Comment:
could we check against the version we got from the checkout instead of against 1?

Reply:
this is explicitly moving from 0, but lets make the check >= 1